### ISSUE TITLE
[fs][v2] scope-wide requireLiteralLeadingDot config not working

### ISSUE THREAD

**Issue by ayangweb** [tauri-apps/plugins-workspace#1927]
The `writeFile` execution reported an error, I suspect that it is because the path is a path containing a `.` path, how can I fix this? I am using `appDataDir` path!

<img width="733" alt="image" src="https://github.com/user-attachments/assets/9893fd77-fa27-42b2-b49d-0270c1b001b5">

**Comment by FabianLars** [comment]
Can you also share your configured permissions?

**Comment by ayangweb** [comment]
```json
{
  "fs:read-files",
  "fs:allow-write-file",
  "fs:allow-copy-file",
  "fs:allow-remove",
  {
    "identifier": "fs:scope",
    "allow": ["**/*"]
  },
}
```

**Comment by ayangweb** [comment]
It works fine in macos and windows! Only in linux it reports errors!

**Comment by FabianLars** [comment]
Yeah, if you only use that scope than the dot is indeed the issue. Either add explicit `$APPDATA` based scopes or change your existing scope to this
```json
{
  "identifier": "fs:scope",
  "allow": ["**/*"],
  "requireLiteralLeadingDot": false
},
```

**Comment by ayangweb** [comment]
Hey👋, `requireLiteralLeadingDot` set to `false` also reports the same error!

https://github.com/user-attachments/assets/4db40efe-5601-4f7d-b2ac-36f8f85c0892

**Comment by FabianLars** [comment]
Hmm, that's weird. Just to double check, does `["$APPDATA/*"]` work? Also, can you share the function call that's failing?

**Comment by ayangweb** [comment]
`["$APPDATA/*"]` also gives an error, also the final export path is chosen by the user, so I wrote `["**/*"]`.

https://github.com/user-attachments/assets/b676c1ec-ee0d-406f-90ee-e32f6ce4dc77

```ts
export const writeFile = (
	path: string,
	data: string,
	options?: WriteFileOptions,
) => {
	const encoder = new TextEncoder();
	const encodeData = encoder.encode(data);

	return tauriWriteFile(path, encodeData, options);
};

await writeFile(getBackupStorePath(), JSON.stringify(content));
```

**Comment by ayangweb** [comment]
On macOS and windows, it is normal

https://github.com/user-attachments/assets/737cfe4e-b4f1-4339-896c-fd60e5c94cd3

**Comment by FabianLars** [comment]
> also the final export path is chosen by the user, so I wrote ["**/*"].

If you use the dialog plugin for that then the scope would be dynamically extended btw, but yeah your solution works too of course.

> On macOS and windows, it is normal

Yeah, that's just another argument for the dot being the issue.

I guess the scope-wide requireLiteralLeadingDot setting doesn't work right now then. Try adding this to tauri.conf.json then:
```json
{
  "plugins": {
    "fs": {
      "requireLiteralLeadingDot": false
    }
  }
}
```

**Comment by ayangweb** [comment]
It's working fine, thanks! This error report should be a problem caused by other code!

https://github.com/user-attachments/assets/1f23f726-8a4b-4e2c-8d5e-9e7a13c2e58d

**Comment by ayangweb** [comment]
By the way, I looked at the source code for the `stat` method, and the size of the folder is also `metadata.len()`, so it doesn't get the correct size, is it designed this way on purpose? Or is it a bug?

https://github.com/tauri-apps/plugins-workspace/blob/7e1c17a635f10d69e10ea9f35081a68c71276203/plugins/fs/src/commands.rs#L1117

**Comment by FabianLars** [comment]
well, it's at least not supposed to get the correct folder size because that only works by iterating over all files recursively which is just too slow for a simple stat function. I don't know whether it should still call len() or just return 0 but as long as it doesn't panic or anything i think it's fine as-is (isn't a folder 4kb or something on Linux?).

**Comment by ayangweb** [comment]
This is the problem I encountered with `stat`, it has nothing to do with the platform, it counts the folder size incorrectly on whichever platform it is on, the folder size needs to be implemented by a recursive function!
